src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java

```diff
@@ -0,0 +1,118 @@
+
+    private final JobLauncher jobLauncher;
+
+    private final EntityManagerFactory entityManagerFactory;
+
+    private final UpdateJurisdictionCodesProcessor updateJurisdictionCodesProcessor;
+
+    @Value("${spring.batch.jurisdictionCodes.pageSize}")
+    private int jurisdictionCodesPageSize;
+
+    @Value("${spring.batch.jurisdictionCodes.maxItemCount}")
+    private int jurisdictionCodesItemCount;
+
+    @Value("${spring.batch.jurisdictionCodes.chunkSize}")
+    private int jurisdictionCodesChunkSize;
+
+    @Autowired
+    public UpdateJurisdictionCodesConfiguration(JobRepository jobRepository,
+                                                PlatformTransactionManager transactionManager, JobLauncher jobLauncher,
+                                                EntityManagerFactory entityManagerFactory,
+                                                UpdateJurisdictionCodesProcessor updateJurisdictionCodesProcessor) {
+        this.jobRepository = jobRepository;
+        this.transactionManager = transactionManager;
+        this.jobLauncher = jobLauncher;
+        this.entityManagerFactory = entityManagerFactory;
+        this.updateJurisdictionCodesProcessor = updateJurisdictionCodesProcessor;
+    }
+
+    @Scheduled(cron = "${spring.batch.jurisdictionCodes.cronJobSchedule}")
+    @SchedulerLock(name = "jurisdictionCodesUpdate")
+    public void scheduleDocumentMetaDataUpdate() throws JobParametersInvalidException,
+        JobExecutionAlreadyRunningException,
+        JobRestartException,
+        JobInstanceAlreadyCompleteException {
+
+        jobLauncher.run(updateDocumentMetaDataJob(), new JobParametersBuilder()
+            .addDate("date", new Date())
+            .toJobParameters());
+    }
+
+    @Bean
+    public LockProvider lockProvider(DataSource dataSource) {
+        return new JdbcTemplateLockProvider(dataSource);
+    }
+
+    public Job updateDocumentMetaDataJob() {
+        return new JobBuilder("updateDocumentMetaDataJob", jobRepository)
+            .flow(new StepBuilder("updateDocumentMetaDataStep",jobRepository)
+                      .<HearingRecordingSegment, HearingRecording>chunk(jurisdictionCodesChunkSize, transactionManager)
+                      .reader(undeletedDocumentsWithTtl())
+                      .processor(updateJurisdictionCodesProcessor)
+                      .writer(itemWriter())
+                      .build()).build().build();
+    }
+
+    private JpaPagingItemReader<HearingRecordingSegment> undeletedDocumentsWithTtl() {
+        return new JpaPagingItemReaderBuilder<HearingRecordingSegment>()
+            .name("documentTaskReader")
+            .entityManagerFactory(entityManagerFactory)
+            .queryString("select hrs from HearingRecordingSegment hrs where hrs.hearingRecording in "
+                             + "(select hr from HearingRecording hr where hr.serviceCode is null and "
+                             + "hr.jurisdictionCode is null order by createdOn asc)")
```

Comment:
is it ok to run this without limit?

Reply:
Limit is not supported in HQL. We are using maxItemCount instead.

Reply:
I meant for inner sql